src/app/dataset-view/additional-components/data-component/data-component.ts

```diff
@@ -96,5 +102,19 @@ export class DataComponent extends BaseComponent implements OnInit {
             // @param editor The editor instance is passed in as a convenience
             run: runQueryFn,
         });
+        this.onRunSQLRequest();
+    }
+
+    private buildSqlRequestCode(): void {
+        if (this.datasetBasics) {
+            this.sqlRequestCode += `'${this.datasetBasics.name as string}'`;
+            const offset = this.location.getState() as Partial<OffsetInterval>;
+            if (
+                typeof offset.start !== "undefined" &&
+                typeof offset.end !== "undefined"
+            ) {
+                this.sqlRequestCode += `\nwhere ${this.offsetColumnName}>=${offset.start} and ${this.offsetColumnName}<${offset.end}\norder by ${this.offsetColumnName} desc\nlimit 50`;
```

Comment:
The interval is closed, so offset = end should be included.
I guess no sense in LIMIT prefix, this is more a pagination subject.

Reply:
Fixed, included.